src/Param.cc

```diff
@@ -403,11 +417,21 @@ bool Param::ValueFromString(const std::string &_value)
              this->dataPtr->typeName == "pose" ||
              this->dataPtr->typeName == "Pose")
     {
-      StringStreamClassicLocale ss(tmp);
-      ignition::math::Pose3d posetmp;
+      // if the value is an empty string or white space we avoid parsing since
+      // its considered a valid value
+      auto isWhiteSpace = [](unsigned char _c)
+      {
+        return std::isspace(_c);
+      };
 
-      ss >> posetmp;
-      this->dataPtr->value = posetmp;
+      if (!tmp.empty() || !std::all_of(tmp.begin(), tmp.end(), isWhiteSpace))
+      {
+        StringStreamClassicLocale ss(tmp);
+        ignition::math::Pose3d posetmp;
+
+        ss >> posetmp;
+        this->dataPtr->value = posetmp;
+      }
```

Comment:
I think the changes to this block are not needed because [Param::SetFromString](https://github.com/osrf/sdformat/blob/stricter_parsing_10/src/Param.cc#L481-L493) already calls `sdf::trim` to remove whitespace and then sets a default value if the trimmed string is empty. I've added tests to confirm this in 9daad39e603fa77987689b03070167b924fd0e3a of branch `scpeters/stricter_parsing_10` after reverting the changes to this block in 8f6120c

Reply:
Sounds good. I see that `ValueFromString` is also called form the constructor without `sdf::trim`, but that should be okay because it's using the `default` string from the spec description.
https://github.com/osrf/sdformat/blob/b80e070dbed613e972acfb81e66b24a248eefd6c/src/Param.cc#L70

Reply:
I just tried it and it looks like `UNIT_Utils_TEST` fails with that change.

Reply:
ok, it sounds like this change is a good idea. Perhaps we could generalize it so it applies to types other than Pose?

Reply:
I added an overload for `sdf::trim` that takes a `std::string` in https://github.com/osrf/sdformat/pull/244/commits/fdecf77b19ee59c062602b2649737b4da2d9a7e4, and used that to trim input values in https://github.com/osrf/sdformat/pull/244/commits/32b3d84ef305ff1f20e09eeab6672dbe6e2d0f68